### ISSUE TITLE
Changes in dialogTemplate are not synced to Snips Console

### ISSUE THREAD

**Issue by glueckself** [project-alice-assistant/ProjectAlice#72]
Hi!

After playing around with the initial implementation of my module I wanted to extend the dialogTemplates. I discovered a bug where not only changes to the JSON files would not be uploaded to the Snips Console, but even deleting the intents or the app in the Console wouldn't solve the problem. `alice sync assistant` says that no changes detected during sync and the refreshing the Snips Console doesn't show them either.

It seems that the cause is the data stored in var/assistant. Removing the proj_... directory, made `alice sync assistant` upload the all intents. While it warns about duplicate intents, the changes are uploaded correctly.

However, this leads to another issue (it should be probably a separate issue): the slots types ("My Slot Types"), the skills ("My Apps") and are uploaded multiple times and added to the same assistant (e.g. My ProjectAlice_en assistant has now 3 AliceCore skills).

**Comment by jr-k** [comment]
Thanks for these details.

Just to be clear, alice can't detect changes if you edit skills on the console. All changes must be edited in local files. Now it seems there is a persistent problem in editing dialogTemplates, I'll try to find the culprit thanks to your message.

**Comment by jr-k** [comment]
Plus it is normal to have multiple skill/slotTypes, only the intents are "moved" because it's too time consuming to take car of these "dead entities" and Snips back-office doesn't notify me with "Skill already exists" or "SlotType already exists", but for intent it is the case.

**Comment by glueckself** [comment]
You're welcome. :)

I deleted the intents in the Console to see if Alice would upload them again, same for the app. The log said that there were no changes for the "mpdPlay" intent (the intent I deleted in the Console before). Only after deleting the stuff in var/assistant would Alice upload the intent.
But yes, I know that this is a one-way sync.

I'm concerned that having multiple entries of the same app in an assistant might cause problems for the Snips Console (or the ASR/NLU training/running). I haven't tried to use such an assistant yet.

I think it might become problematic after some time when the number of skills and slotTypes in the Console get into thousands, because every upload only creates new skills and slotTypes with the same name. If there is an API to query and delete items (skills and slotTypes), it might be better if Alice could try to query an item, and if it returns data, compare it with the data generated from the module. If it matches, then that item is OK, if it doesn't match, then Alice can delete that item and upload the new version.
(I hope I'm not rambling incomprehensibly here ^^) 

Please let me know if you need any details or if I can help somehow (or if you have trouble reproducing it - maybe it was some other mistake on my end that caused it ^^).
I'll see if I find something in the code (and get familiar with it :) )

**Comment by glueckself** [comment]
I hacked together a script to mass delete assistants, skills, intents and entities in the Snips Console: https://gist.github.com/glueckself/f2243b17dd3c87c9ac3d14bb644bc8a1.

I also created a second Snips account just for Alice so that I can simply delete everything.

**Comment by Psychokiller1888** [comment]
Good stuff!
